src/store.rs

```diff
@@ -14,20 +14,75 @@
  * limitations under the License.
  */
 
+use rocket::futures::TryStreamExt;
+use rocket_db_pools::mongodb::options::{FindOptions, InsertOneOptions};
+use rocket_db_pools::mongodb::Client;
+use rocket_db_pools::mongodb::error::Error;
+use rocket_db_pools::mongodb::bson::doc;
+use rocket_db_pools::mongodb::results::InsertOneResult;
+use rocket_db_pools::mongodb::Collection;
+use rocket::serde::{Deserialize, Serialize};
+
+
 use murmur::MurmurStore;
-use std::fs::File;
+const DB_NAME: &str = "MurmurDB";
+const DB_URI: &str = "mongodb+srv://murmurapi:GuVsTAEbQtNnnbPj@useast.m8j6h.mongodb.net/?retryWrites=true&w=majority&appName=USEast";
+const COLLECTION_NAME: &str = "mmrs";
 
-pub(crate) fn load() -> MurmurStore {
-	// TODO: load from DB
-	let mmr_store_file = File::open("mmr_store").expect("Unable to open file");
-	let data: MurmurStore = serde_cbor::from_reader(mmr_store_file).unwrap();
+#[derive (Serialize, Deserialize)]
+pub struct MurmurDbObject {
+	pub mmr: MurmurStore,
+	pub username: String
+}
 
-	data
+pub(crate) struct Store {
+	pub(crate) col: Collection<MurmurDbObject>,
 }
 
-/// Write the MMR data to a file
-pub(crate) fn write(mmr_store: MurmurStore) {
-	// TODO: write to DB
-	let mmr_store_file = File::create("mmr_store").expect("It should create the file");
-	serde_cbor::to_writer(mmr_store_file, &mmr_store).unwrap();
+impl Store {
+	pub(crate) async fn init() -> Self {
+		let client = Client::with_uri_str(DB_URI).await.unwrap();
+		let col = client.database(&DB_NAME).collection(&COLLECTION_NAME);
+		Store { col }
+	}
+
+	pub(crate) async fn load(
+		&self,
+		username: &str,
+		options: Option<FindOptions>,
+	) -> Result<Option<MurmurStore>, Error> {
+	
+		let filter = doc! {"username": username};
+		// let mmr_collection: Collection<MurmurDbObject> = db.database(&DB_NAME).collection(&COLLECTION_NAME);
+		let cursor_result = self.col.find(filter, options).await;
+	
+		match cursor_result {
+			Err(e) => Err(e),
+			Ok(mut mmr_cursor) => {
+				let mmr_next = mmr_cursor.try_next().await;
+				match mmr_next {
+					Err(e) => Err(e),
+					Ok(mmr_option) => {
+						match mmr_option {
+							Some(mmr_db_object) => {
+								Ok(Some(mmr_db_object.mmr))
+							},
+							None => Ok(None)
+						}
+					}
+				}
+			}
+		}
```

Comment:
When we don't need to transform the Error from a Result, we can just use the `?` operator. The question mark would unwrap if Ok or return the Error. This saves us the nested conditionals, and several lines of code:
```suggestion
		let mut mmr_cursor = self.col.find(filter, options).await?;

		let mmr_option = mmr_cursor.try_next().await?;
		match mmr_option {
			Some(mmr_db_object) => Ok(Some(mmr_db_object.mmr)),
			None => Ok(None),
		}
```